File: hw/cevallos/u3/hw35StrategyPersistence/JavaScript/calendar-system/controller/EventController.js
const Event = require('../model/Event');
const StrategyFactory = require('../factory/StrategyFactory');

class EventController {
    constructor(storageType = 'json') {
        this.storageType = storageType;
        this.strategy = StrategyFactory.createStrategy(storageType);
    }

    setStorageStrategy(storageType) {
        this.storageType = storageType;
        this.strategy = StrategyFactory.createStrategy(storageType);
        console.log(`Estrategia cambiada a: ${storageType}`);
    }

    async addEvent(id, name, date) {
        try {
            const event = new Event();
            event.id = id;
            event.name = name;
            event.date = date;
            
            const success = await this.strategy.addEvent(event);
            return { success, event };
        } catch (error) {
            return { success: false, error: error.message };
        }
    }

    async updateEvent(id, name, date) {
        try {
            const event = new Event();
            event.id = id;
            event.name = name;
            event.date = date;
            
            const success = await this.strategy.updateEvent(event);
            return { success, event };
        } catch (error) {
            return { success: false, error: error.message };
        }
    }

    async deleteEvent(id) {
        try {
            const success = await this.strategy.deleteEvent(id);
            return { success };
        } catch (error) {
            return { success: false, error: error.message };
        }
    }

    async readEvent(id) {
        try {
            const event = await this.strategy.readEvent(id);
            return { success: !!event, event };
        } catch (error) {
            return { success: false, error: error.message };
        }
    }

    async getAllEvents() {
        try {
            const events = await this.strategy.getAllEvents();
            return { success: true, events };
        } catch (error) {
            return { success: false, error: error.message, events: [] };
        }
    }

    createEventObject(id, name, date) {
        try {
            const event = new Event();
            event.id = id;
            event.name = name;
            event.date = date;
            return { success: true, event };
        } catch (error) {
            return { success: false, error: error.message };
        }
    }
}

module.exports = EventController;

File: hw/cevallos/u3/hw35StrategyPersistence/Python/calendar_app/controller/event_controller.py
from typing import Optional
from model.event import Event
from strategy.storage_strategy import StorageStrategy
from strategy.factory import StorageStrategyFactory

class EventController:
    def __init__(self, storage_type: str = "json"):
        self.storage_type = storage_type
        self.strategy = StorageStrategyFactory.create_strategy(storage_type)
    
    def set_storage_strategy(self, storage_type: str):
        """Cambiar la estrategia de almacenamiento"""
        self.storage_type = storage_type
        self.strategy = StorageStrategyFactory.create_strategy(storage_type)
    
    def add_event(self, event: Event) -> bool:
        return self.strategy.add_event(event)
    
    def update_event(self, event: Event) -> bool:
        return self.strategy.update_event(event)
    
    def delete_event(self, event_id: str) -> bool:
        return self.strategy.delete_event(event_id)
    
    def read_event(self, event_id: str) -> Optional[Event]:
        return self.strategy.read_event(event_id)
    
    def create_event(self, event_id: str, name: str, date: str) -> Event:
        """Método helper para crear un evento"""
        event = Event()
        event.id = event_id
        event.name = name
        event.date = date
        return event